### PR TITLE
Add logging on map-target access

### PR DESCRIPTION
**Pull request by svenwest**
Log on level 2 if access fails, level 4 otherwise

### CODE REVIEW

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581007

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581009

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581638

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581637

File: lib/1.4/utility.dml [hunk 1]
```diff
@@ -1371,6 +1371,12 @@ template map_target is (connect, _qname) {
                 return Sim_PE_IO_Not_Taken;
             }
         }
-        return SIM_issue_transaction(map_target, t, addr);
+        local exception_type_t exc = SIM_issue_transaction(map_target, t, addr);
+        local bool fail = exc != Sim_PE_No_Exception;
+        log info, (fail ? 2 : 4) : "%s%s %d bytes @ 0x%x in %s",
+            fail ? "failed to " : "",
+            SIM_transaction_is_read(t) ? "read" : "write",
```
Comment (mandolaerik) [inline lib/1.4/utility.dml:1378]: Inconsistent grammar: `{failed to read, failed to writre, read} N bytes` all qualify as past tense, whereas `write N bytes` does not.

Comment (svenwest) [reply lib/1.4/utility.dml:1378]: Got it, I'll change `failed => fail` and add an exclamation mark, making the grammar consistently imperative tense ;-)

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9584738

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9584744

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9584743

File: lib/1.4/utility.dml (hunk 1 above)
Comment (mandolaerik) [reply lib/1.4/utility.dml]: It seems that this change didn't make it into your latest push.

I would advise against an exclamation mark; we don't use punctuation in short messages and an exclamation mark makes the tone less neutral. (Also I'm not *against* past tense when you describe something in the past; my problem was with the inconsistency. I don't know if there's a recommendation on past vs imperative here)

Comment (svenwest) [reply lib/1.4/utility.dml]: Sorry, the comment was a joke. My latest push changes the grammar to consistently past tense.
> read 8 bytes @ 0x0 in phys_mem
> wrote 8 bytes @ 0x0 in phys_mem
> failed to read 8 bytes @ 0x0 in phys_mem
> failed to write 8 bytes @ 0x0 in phys_mem

```
        log info, (fail ? 2 : 4) : "%s%s %d bytes @ 0x%x in %s",
            fail ? "failed to " : "",
            SIM_transaction_is_read(t) ? "read" : fail ? "write" : "wrote",
            SIM_transaction_size(t), addr, SIM_object_name(obj);
        return exc;
```

**Review by mandolaerik — APPROVED** [review]

**Pull request merged**